static/app/components/replays/diff/replaySliderDiff.tsx

```diff
@@ -0,0 +1,166 @@
+import {Fragment, useRef} from 'react';
+import styled from '@emotion/styled';
+
+import NegativeSpaceContainer from 'sentry/components/container/negativeSpaceContainer';
+import ReplayIFrameRoot from 'sentry/components/replays/player/replayIFrameRoot';
+import {StaticReplayPreferences} from 'sentry/components/replays/preferences/replayPreferences';
+import {Provider as ReplayContextProvider} from 'sentry/components/replays/replayContext';
+import {space} from 'sentry/styles/space';
+import toPixels from 'sentry/utils/number/toPixels';
+import type ReplayReader from 'sentry/utils/replays/replayReader';
+import {useDimensions} from 'sentry/utils/useDimensions';
+import {useResizableDrawer} from 'sentry/utils/useResizableDrawer';
+
+interface Props {
+  leftOffsetMs: number;
+  replay: null | ReplayReader;
+  rightOffsetMs: number;
+}
+
+export function ReplaySliderDiff({leftOffsetMs, replay, rightOffsetMs}: Props) {
+  const positionedRef = useRef<HTMLDivElement>(null);
+  const viewDimensions = useDimensions({elementRef: positionedRef});
+
+  const width = toPixels(viewDimensions.width);
+  return (
+    <WithPadding>
+      <Positioned ref={positionedRef}>
+        {viewDimensions.width ? (
```

Comment:
Is this what we want if width is 0?

Reply:
i think so? up until this morning it was easy/common to get `0` during the first render, but now we have a `useLayoutEffect` happening https://github.com/getsentry/sentry/pull/73958

I think if the width is 0, then it's not visible and we can skip inserting the children, it should re-render when the size is increased.